src/adagrams.js

```diff
@@ -1,9 +1,72 @@
+// import { keys } from "core-js/core/array";
+
+// import { forEach } from "core-js/core/array";
+
+const letterBank = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const lettersArr = Object.keys(letterBank);
+
+  let letterPoolFreq = {};
+  let letterPool = [];
+
+  while (letterPool.length < 10) {
+    let randomIndex = Math.floor(Math.random() * lettersArr.length);
+    let randLetter = lettersArr[randomIndex];
+
+    if (!(randLetter in letterPoolFreq)){ // this is checking that if random letter not in letterpool it's frequency now = 1
```

Comment:
This logic ensures that a letter can't be used more times than is allowed by the letterBank, so the tests pass; however, the distribution put forth in the letterBank is still not being used in generating a random letter. The every letter you draw is always a 1/26 chance, rather than, e.g. for the letter T, a 6/98 chance.

Reply:
Ah I see what you are saying, I will rework it with the weighted number logic tomorrow!